### PR TITLE
Scripts/Commands: Modify fury & pain

### PR DESCRIPTION
**Pull request by Hadifaren**
**Changes proposed:**

- Modify some files to edit Fury & Pain
- class Demon Hunter

**Target branch(es):** 3.3.5/master
- [X] master

**Tests performed:** (Does it build, tested in-game, etc.)
- [X] build successful
- [X] tested in game

**Known issues and TODO list:** (add/remove lines as needed)

### CODE REVIEW

**Comment by DDuarte** [comment]
It looks better than the previous PR 👍 

But please, change tabs to 4 spaces and you can use RBAC permissions ids below 1000

File: src/server/scripts/Commands/cs_modify.cpp [hunk 1]
```diff
@@ -1213,9 +1216,41 @@ class modify_commandscript : public CommandScript
...
+		return false;
+	}
+		
 };
 
 void AddSC_modify_commandscript()
 {
     new modify_commandscript();
-}
+}
```
Comment (Seyden) [inline src/server/scripts/Commands/cs_modify.cpp]: newline

File: src/server/game/Miscellaneous/Language.h [hunk 2]
```diff
@@ -1242,6 +1242,10 @@ enum TrinityStrings
...
     LANG_SHUTDOWN_DELAYED                         = 11017,
-    LANG_SHUTDOWN_CANCELLED                       = 11018
+    LANG_SHUTDOWN_CANCELLED                       = 11018,
+	LANG_YOU_CHANGE_FURY                          = 11019,
+	LANG_YOUR_FURY_CHANGED                        = 11020,
+	LANG_YOU_CHANGE_PAIN                          = 11021,
+	LANG_YOUR_PAIN_CHANGED                        = 11022	
 };
-#endif
+#endif
```
Comment (Seyden) [inline src/server/game/Miscellaneous/Language.h]: newline

File: src/server/game/Accounts/RBAC.h [hunk 3]
```diff
@@ -1021,4 +1023,4 @@ class TC_GAME_API RBACData
 
 }
 
-#endif
+#endif
```
Comment (Seyden) [inline src/server/game/Accounts/RBAC.h]: newline

Comment (Hadifaren) [inline src/server/game/Accounts/RBAC.h]: what you mean? whats wrong

Comment (DDuarte) [inline src/server/game/Accounts/RBAC.h]: Simply add an empty line after the end of the file

Comment (Hadifaren) [inline src/server/game/Accounts/RBAC.h]: added empty lines please check now

**Comment by Shauren** [comment]
Why don't we have a single modify power command that takes power type as argument?

File: src/server/game/Accounts/RBAC.h [hunk 4]
```diff
@@ -759,6 +759,8 @@ enum RBACPermissions
     RBAC_PERM_COMMAND_GO_OFFSET                              = 852,
 
     // custom permissions 1000+
+	RBAC_PERM_COMMAND_MODIFY_FURY                            = 1001,
```
Comment (Aokromes) [inline src/server/game/Accounts/RBAC.h]: tabs.

File: src/server/game/Miscellaneous/Language.h [hunk 5]
```diff
@@ -1242,6 +1242,10 @@ enum TrinityStrings
     LANG_CREATURE_NOT_AI_ENABLED                  = 11015,
     LANG_SELECT_PLAYER_OR_PET                     = 11016,
     LANG_SHUTDOWN_DELAYED                         = 11017,
-    LANG_SHUTDOWN_CANCELLED                       = 11018
+    LANG_SHUTDOWN_CANCELLED                       = 11018,
+	LANG_YOU_CHANGE_FURY                          = 11019,
```
Comment (Aokromes) [inline src/server/game/Miscellaneous/Language.h]: tabs

File: src/server/scripts/Commands/cs_modify.cpp [hunk 6]
```diff
@@ -70,6 +70,9 @@ class modify_commandscript : public CommandScript
             { "standstate",   rbac::RBAC_PERM_COMMAND_MODIFY_STANDSTATE,   false, &HandleModifyStandStateCommand,    "" },
             { "talentpoints", rbac::RBAC_PERM_COMMAND_MODIFY_TALENTPOINTS, false, &HandleModifyTalentCommand,        "" },
             { "xp",           rbac::RBAC_PERM_COMMAND_MODIFY_XP,           false, &HandleModifyXPCommand,            "" },
+			{ "fury",         rbac::RBAC_PERM_COMMAND_MODIFY_FURY,         false, &HandleModifyFuryCommand,          "" },
```
Comment (Aokromes) [inline src/server/scripts/Commands/cs_modify.cpp]: tabs

File: src/server/scripts/Commands/cs_modify.cpp [hunk 7]
```diff
@@ -1213,6 +1216,38 @@ class modify_commandscript : public CommandScript
         target->GiveXP(xp, nullptr);
         return true;
     }
+	//Edit Player Fury
```
Comment (Aokromes) [inline src/server/scripts/Commands/cs_modify.cpp]: tabs.

File: src/server/game/Accounts/RBAC.h (hunk 4 above)
Comment (Hadifaren) [inline src/server/game/Accounts/RBAC.h]: correct
no problem in Visual studio, i dont know how to fix this tab Oo

File: src/server/scripts/Commands/cs_modify.cpp (hunk 6 above)
Comment (Hadifaren) [inline src/server/scripts/Commands/cs_modify.cpp]: cs_modify

visual studio showing correct

Comment (DDuarte) [inline src/server/scripts/Commands/cs_modify.cpp]: http://lmgtfy.com/?q=visual+studio+change+tabs+to+spaces 🙂

File: src/server/game/Accounts/RBAC.h (hunk 4 above)
Comment (Rochet2) [inline src/server/game/Accounts/RBAC.h]: http://stackoverflow.com/questions/14167033/visual-studio-replace-tab-with-4-spaces
google.

Comment (boxa) [inline src/server/game/Accounts/RBAC.h]: ёпрст! MSVS 2015:

image

**Comment by Hadifaren** [comment]
Thank you all for helping check it now :P

File: src/server/scripts/Commands/cs_modify.cpp [hunk 8]
```diff
@@ -70,6 +70,9 @@ class modify_commandscript : public CommandScript
             { "standstate",   rbac::RBAC_PERM_COMMAND_MODIFY_STANDSTATE,   false, &HandleModifyStandStateCommand,    "" },
             { "talentpoints", rbac::RBAC_PERM_COMMAND_MODIFY_TALENTPOINTS, false, &HandleModifyTalentCommand,        "" },
             { "xp",           rbac::RBAC_PERM_COMMAND_MODIFY_XP,           false, &HandleModifyXPCommand,            "" },
+            { "fury",         rbac::RBAC_PERM_COMMAND_MODIFY_FURY,         false, &HandleModifyFuryCommand,          "" },
+            { "pain",         rbac::RBAC_PERM_COMMAND_MODIFY_PAIN,         false, &HandleModifyPainCommand,          "" },
+
```
Comment (offl) [inline src/server/scripts/Commands/cs_modify.cpp]: Excess line.

File: src/server/game/Miscellaneous/Language.h [hunk 9]
```diff
@@ -1242,6 +1242,10 @@ enum TrinityStrings
     LANG_CREATURE_NOT_AI_ENABLED                  = 11015,
     LANG_SELECT_PLAYER_OR_PET                     = 11016,
     LANG_SHUTDOWN_DELAYED                         = 11017,
-    LANG_SHUTDOWN_CANCELLED                       = 11018
+    LANG_SHUTDOWN_CANCELLED                       = 11018,
+    LANG_YOU_CHANGE_FURY                          = 11019,
+    LANG_YOUR_FURY_CHANGED                        = 11020,
+    LANG_YOU_CHANGE_PAIN                          = 11021,
+    LANG_YOUR_PAIN_CHANGED                        = 11022
```
Comment (Rochet2) [inline src/server/game/Miscellaneous/Language.h]: looks like there is trailing whitespace here `= 11022	`

File: src/server/scripts/Commands/cs_modify.cpp [hunk 10]
```diff
@@ -70,6 +70,8 @@ class modify_commandscript : public CommandScript
             { "standstate",   rbac::RBAC_PERM_COMMAND_MODIFY_STANDSTATE,   false, &HandleModifyStandStateCommand,    "" },
             { "talentpoints", rbac::RBAC_PERM_COMMAND_MODIFY_TALENTPOINTS, false, &HandleModifyTalentCommand,        "" },
             { "xp",           rbac::RBAC_PERM_COMMAND_MODIFY_XP,           false, &HandleModifyXPCommand,            "" },
+            { "fury",         rbac::RBAC_PERM_COMMAND_MODIFY_FURY,         false, &HandleModifyFuryCommand,          "" },
+            { "pain",         rbac::RBAC_PERM_COMMAND_MODIFY_PAIN,         false, &HandleModifyPainCommand,          "" },
```
Comment (Rochet2) [inline src/server/scripts/Commands/cs_modify.cpp]: trailing whitespace

File: src/server/scripts/Commands/cs_modify.cpp [hunk 11]
```diff
@@ -1213,6 +1215,36 @@ class modify_commandscript : public CommandScript
...
+        int8 const furyMultiplier = 10;
+        if (CheckModifyResources(handler, args, target, fury, furymax, furyMultiplier))
+        {
+            NotifyModification(handler, target, LANG_YOU_CHANGE_FURY, LANG_YOUR_FURY_CHANGED, fury / furyMultiplier, furymax / furyMultiplier);
+            target->SetMaxPower(POWER_FURY, furymax);
+            target->SetPower(POWER_FURY, fury);
+            return true;
+        }
+        return false;
+    }
```
Comment (Rochet2) [inline src/server/scripts/Commands/cs_modify.cpp]: trailing whitespace

File: src/server/scripts/Commands/cs_modify.cpp [hunk 12]
```diff
@@ -1213,6 +1215,36 @@ class modify_commandscript : public CommandScript
...
+        int8 const painMultiplier = 10;
+        if (CheckModifyResources(handler, args, target, pain, painmax, painMultiplier))
+        {
+            NotifyModification(handler, target, LANG_YOU_CHANGE_PAIN, LANG_YOUR_PAIN_CHANGED, pain / painMultiplier, painmax / painMultiplier);
+            target->SetMaxPower(POWER_PAIN, painmax);
+            target->SetPower(POWER_PAIN, pain);
+            return true;
+        }
+        return false;
+    }
```
Comment (Rochet2) [inline src/server/scripts/Commands/cs_modify.cpp]: and more whitespace at end of line

**Comment by tkrokli** [comment]
Please also synchronize (rebase) your PR branch to get rid of old errors (outside of your PR) in the TC source.

**Comment by Aokromes** [comment]
@Hadifaren read the previous comment :)

**Comment by Hadifaren** [comment]
Hi , i have created a new one to modify every power 
so please someone modify this pull as i cant do it my self my core is 7.1.5 so you may need to change numbers too
https://gist.github.com/Hadifaren/08a3190138fcd8d866689bcf1f75f13c/revisions

**Comment by Hadifaren** [comment]
updated : https://gist.github.com/Hadifaren/e54b8b7f44d9d2d0a6924b885775fdb6

**Comment by tkrokli** [comment]
Because you deleted your PR branch, it is no longer possible for any of us to update this PR.
It is better if you open a new PR and write
**Closes #19288**
in the new PR description template.

**Comment by tkrokli** [comment]
By the way: please make sure your `master` branch is up to date before you start adding your PR content in the new branch (branched off from the updated master).

**Comment by Hadifaren** [comment]
@tkrokli  
Check this please #20833

**Pull request closed without merging**